recipes/nuraft/all/patches/0001-cmake_patches.patch

```diff
@@ -1,24 +1,35 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
-index 87685ff..1fc9199 100644
+index 87685ff..efbc7fb 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -26,41 +44,41 @@ endif()
+@@ -1,5 +1,6 @@
+ cmake_minimum_required(VERSION 3.5)
+ project(NuRaft VERSION 1.0.0 LANGUAGES CXX)
++set (CMAKE_CXX_STANDARD 11)
+ 
+ # === Build type (default: RelWithDebInfo, O2) ===========
+ if (NOT CMAKE_BUILD_TYPE)
+@@ -26,41 +27,23 @@ endif()
  
  
  # === Find ASIO ===
 -if (BOOST_INCLUDE_PATH AND BOOST_LIBRARY_PATH)
--    # If Boost path (both include and library) is given,
--    # use Boost's ASIO.
++find_package(OpenSSL CONFIG REQUIRED)
++find_package(Boost CONFIG)
++if (Boost_FOUND)
+     # If Boost path (both include and library) is given,
+     # use Boost's ASIO.
 -    message(STATUS "Boost include path: " ${BOOST_INCLUDE_PATH})
 -    message(STATUS "Boost library path: " ${BOOST_LIBRARY_PATH})
 -
--    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_BOOST_ASIO")
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_BOOST_ASIO")
 -
 -    set(ASIO_INCLUDE_DIR ${BOOST_INCLUDE_PATH})
 -    set(LIBBOOST_SYSTEM "${BOOST_LIBRARY_PATH}/libboost_system.a")
 -
--else ()
--    # If not, ASIO standalone mode.
++    set(ASIO_DEP boost::boost)
```

Comment:
I noticed this was replaced below 🤔 

Should we not use the asio target?

Reply:
I think it is? If `Boost_FOUND` tests false I find the asio package and set ASIO_DEP to `asio::asio` and use `${ASIO_DEP}` as the target_link_library. Am I missing the point?

Reply:
I thought there was `Boost::asio` but it's not official now that I double check https://cmake.org/cmake/help/latest/module/FindBoost.html#imported-targets